controllers/drcluster_controller.go

```diff
@@ -505,20 +498,37 @@ func (u *drclusterInstance) statusConditionSetAndUpdate(
 }
 
 func (u *drclusterInstance) statusUpdate() error {
-	if !reflect.DeepEqual(u.savedInstanceStatus, u.object.Status) {
```

Comment:
We do not require this change in code (only in the tests). If the reconcile fails to update the status, a subsequent reconcile will update the same on the latest version of the resource.

As the intention here is to update the status if it differs as per the current resource under reconcile. If we fetch the latest resource it may have other spec updates that we have not processed and hence we may reflect incorrect status for the same.(for example we reconciled generation 2 of the resource, and we fail to update the status, but as part of the new code fetch generation 3 of the resource and update the status as per generation 2 processing that is done)

Reply:
Yes, I can see the point.

Reply:
@ShyamsundarR , thank you for the explanation. I remove the code, protecting drcluster status update. Hopefully, the next Reconcile will solve the problem.